src/app/features/course-info/course-info.component.ts

```diff
@@ -4,64 +4,64 @@ import { ButtonLabels } from "@app/shared/constants/button-labels";
 import { Location } from "@angular/common";
 import { CoursesStoreService } from "@app/services/courses-store.service";
 import { Course } from "@app/shared/models/course.model";
-import { forkJoin, Subscription } from "rxjs";
+import { map, Observable, Subscription } from "rxjs";
+import { CoursesStateFacade } from "@app/store/courses/courses.facade";
 
 @Component({
   selector: "app-course-info",
   templateUrl: "./course-info.component.html",
   styleUrls: ["./course-info.component.scss"],
 })
 export class CourseInfoComponent implements OnInit, OnDestroy {
-  course!: Course | undefined;
+  course!: Course | null;
+  course$: Observable<Course | null> = this.coursesStateFacade.course$;
   btnTextBack: string = ButtonLabels.back;
   courseId: string | null = null;
-  courseAuthors!: string[];
   authorsByName: string[] = [];
-  private subscriptions: Subscription = new Subscription();
+  private subscriptions: Subscription[] = [];
 
   constructor(
     private route: ActivatedRoute,
     private location: Location,
-    private courseStoreService: CoursesStoreService
+    private courseStoreService: CoursesStoreService,
+    private coursesStateFacade: CoursesStateFacade
   ) {}
 
   ngOnInit(): void {
     const courseId = this.route.snapshot.paramMap.get("id");
 
     if (courseId) {
-      this.courseStoreService.getCourse(courseId).subscribe({
-        next: (course) => {
-          this.course = course.result;
-          this.courseAuthors = course.result.authors;
-          if (this.courseAuthors.length > 0) {
-            this.createAuthorsNameList(this.courseAuthors);
+      this.coursesStateFacade.getSingleCourse(courseId);
```

Comment:
This needs some refactoring. Subscribing within another in 3 levels of depth is not clean and efficient. Hint: combine the subscriptions however it is needed.